esp-hal/src/timer/systimer.rs

```diff
@@ -83,30 +23,27 @@ use crate::{
     peripherals::{Interrupt, SYSTIMER},
     sync::{lock, Lock},
     system::{Peripheral as PeripheralEnable, PeripheralClockControl},
-    Async,
-    Blocking,
     Cpu,
     InterruptConfigurable,
-    Mode,
 };
 
 /// System Timer driver.
 pub struct SystemTimer<'d> {
     /// Unit 0
-    pub unit0: SpecificUnit<'d, 0>,
+    pub unit0: AnyUnit<'d>,
```

Comment:
Any reason not to keep the specific ones here and accept `impl Into<AnyUnit>` in `Alarm::new(....)`?

Reply:
I've completely removed the `Specific*` types in favour of the `Any*` types, I've inlined the functionality from the traits to the `Any*` types.

I've also removed `FrozenUnit`, in favour of a one time configuration when creating `Systimer`.

Reply:
So with these new changes, there's no way to take exclusive ownership of a unit. This is particularly important when you're resetting the counter on it, as you want to make sure nothing else is relying on the count.
And with the `Specific*` types gone, theft, which was useful for interrupts, is no longer possible,
But I guess the thinking here is, expose basic features, if users want something more complex or different guarantees, they should write it themselves using the PAC.

At this point you may as well just remove the `Unit` enum and simplify the driver further by making all the alarms use unit 0. Without the ownership semantics there's no point exposing the concept of a unit to the user at all. Then you can just merge `Comparator` and `Alarm` into the same struct. Vastly simplifying the systimer driver.